### PR TITLE
update initial route documentation

### LINKED ISSUE #57101: Navigator initialRoute documentation comments misleading
The documentation comments above `initialRoute` inside the `Navigator` widget say
```dart
/// If this string contains any `/` characters, then the string is split on 
/// those characters and substrings from the start of the string up to each 
/// such character are, in turn, used as routes to push.
```
Which would seem to indicate that if I specify an initialRoute such as `foo/baz/123`, that the Navigator would push routes `foo`, `foo/baz` and `foo/baz/123` onto the stack. Yet, the only route that gets generated is the one for `foo/baz/123`, the entire thing.

This behavior made more sense after reading the code for `NavigatorState.initState`
```dart
String initialRouteName = widget.initialRoute ?? Navigator.defaultRouteName;
if (initialRouteName.startsWith('/') && initialRouteName.length > 1) {
  initialRouteName = initialRouteName.substring(1); // strip leading '/'
  assert(Navigator.defaultRouteName == '/');
  // ...
} else {
  // ...
```

### PR DESCRIPTION
**Pull request by chunhtai**
## Description

fix the documentation of initial route

## Related Issues

Fixes https://github.com/flutter/flutter/issues/57101

## Tests

no test needed

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [ ] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/navigator.dart [hunk 1]
```diff
@@ -2151,9 +2151,10 @@ class Navigator extends StatefulWidget {
   /// This is the default value of [onGenerateInitialRoutes], which is used if
   /// [initialRoute] is not null.
   ///
-  /// If this string contains any `/` characters, then the string is split on
-  /// those characters and substrings from the start of the string up to each
-  /// such character are, in turn, used as routes to push.
+  /// If this string starts with a `/` character and has multiple `/` characters
+  /// in it, then the string is split on those characters and substrings from
+  /// the start of the string up to each such character are, in turn, used as
+  /// routes to push.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/navigator.dart:2157]: This long sentence was a little confusing. Does this mean each substring split by `/` is treated as an individual route?

**Review by Piinks — APPROVED** [review]
LGTM, if I am understanding the sentence below correctly. :)

File: packages/flutter/lib/src/widgets/navigator.dart (hunk 1 above)
Comment (chunhtai) [reply packages/flutter/lib/src/widgets/navigator.dart:2157]: The substrings from the start of the string up to each `/`is treated as individual routes. The next paragraph describes an example. I am not sure to better describe it.

Comment (Piinks) [reply packages/flutter/lib/src/widgets/navigator.dart:2157]: Ok! SGTM

**Pull request merged**